kratos/input_output/vtu_output.h

```diff
@@ -0,0 +1,248 @@
+     * Hence, the new expression takes place. Therefore, when container expressions required to be outputted, then it is best
+     * to always clear the existing container expressions and add the new ones. Otherwise, the vtu output may be writing not the
+     * latest container expression.
+     *
+     * @tparam TContainerType
+     * @param rExpressionName           Name for the container expression.
+     * @param pContainerExpression      Container expression.
+     */
+    template <class TContainerType>
+    void AddContainerExpression(
+        const std::string& rExpressionName,
+        const typename ContainerExpression<TContainerType>::Pointer pContainerExpression);
+
+    /**
+    * @brief Clears the historical variables.
+    */
+    void ClearHistoricalVariables();
+
+    /**
+    * @brief Clears the nodal non-historical variables.
+    */
+    void ClearNodalNonHistoricalVariables();
+
+    /**
+    * @brief Clears the cell non-historical variables.
+    */
+    void ClearCellNonHistoricalVariables();
+
+    /**
+    * @brief Clears the nodal flags.
+    */
+    void ClearNodalFlags();
+
+    /**
+    * @brief Clears the cell flags.
+    */
+    void ClearCellFlags();
+
+    /**
+    * @brief Clears the nodal container expressions.
+    */
+    void ClearNodalContainerExpressions();
+
+    /**
+    * @brief Clears the cell container expressions.
+    */
+    void ClearCellContainerExpressions();
+
+    /**
+    * @brief Returns the model part.
+    * @return The constant reference to the model part.
+    */
+    const ModelPart& GetModelPart() const;
+
+    /**
+     * @brief Writes the Vtu file.
+     * @details This writes the final vtu file. If this is a transient output, then @ref rOutputFilenamePrefix
+     * should have indication of the step, otherwise this will overwrite the same file.
+     * In the MPI case, this will create one .vtu file per rank, and a .pvtu file to combine them.
+     * @param rOutputFilenamePrefix         Output file name prefix.
+     */
+    void PrintOutput(const std::string& rOutputFilenamePrefix);
+
+    ///@}
+private:
+    ///@name Private member variables
+    ///@{
+
+    ModelPart& mrModelPart; /// Reference to the model part.
+
+    const bool mIsInitialConfiguration; /// Flag indicating if it is the initial configuration.
+
+    const WriterFormat mOutputFormat; /// The output format for writing the model part.
+
+    const IndexType mPrecision; /// The precision used for writing floating-point values.
+
+    bool mIsConditionsConsidered; /// Flag indicating if conditions are considered.
+
+    bool mIsElementsConsidered; /// Flag indicating if elements are considered.
+
+    std::unordered_map<IndexType, IndexType> mKratosVtuIndicesMap; /// Map to store Kratos VTU indices.
+
+    std::unordered_map<std::string, SupportedVariables> mHistoricalVariablesMap; /// Map to store supported historical variables.
```

Comment:
After thinking, I will block, is more efficient to use the IDs of the variable. Then if you want to know the corresponding name of the variable you can store the variables names with the corresponding ID, but string search is quite costly

Reply:
The `std::string` key is only used to cross check whether a `ContainerExpression` or any other (like Flags) are stored with the same name and throw an error.  That means, when adding a variable/flag or container expression. When writing, `std::unordered_map::find` is not used, all the values are written. Hence do you think it still make sense to change to variable id?

If I change only this to the variable id, then 
https://github.com/KratosMultiphysics/Kratos/blob/fde531a34f14ff8da8d1e7179f5cce6b0a5768ba/kratos/input_output/vtu_output.cpp#L363-L370

This needs to be changed with lots of template gymnastics. What is your opinion? I would not mind this extra cost since this is may be called 20 times (adding 20 vars) per single simulation.

Reply:
I am sure it is possible to rearrange using keys instead of strings, but I can live with it

Reply:
I would add a TODO, to check this in the future. We will discuss then.